ci/terraform/modules/lambda-role-policy-reduction/policy.tf

```diff
@@ -0,0 +1,73 @@
+data "aws_iam_policy_document" "logging_policy_document" {
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "logs:CreateLogStream",
+      "logs:PutLogEvents"
+    ]
+
+    resources = [
+      "arn:aws:logs:*:*:*",
+    ]
+  }
+}
+
+data "aws_iam_policy_document" "endpoint_xray_policy" {
+  statement {
+    effect = "Allow"
+    actions = [
+      "xray:*"
+    ]
+
+    resources = [
+      "*",
+    ]
+  }
+}
+
+data "aws_iam_policy_document" "networking_policy_document" {
+  count = var.vpc_arn == "" ? 0 : 1
+  statement {
+    effect = "Allow"
+
+    actions = [
+      "ec2:DescribeNetworkInterfaces",
+      "ec2:CreateNetworkInterface",
+      "ec2:DeleteNetworkInterface"
+    ]
+    resources = [
+      "*"
+    ]
+    condition {
+      test     = "ArnLikeIfExists"
+      variable = "ec2:Vpc"
+      values   = [var.vpc_arn]
+    }
+  }
+}
+
+locals {
+  policy_documents_to_attach = merge({
+    base = concat([
+      data.aws_iam_policy_document.logging_policy_document.json,
+      data.aws_iam_policy_document.endpoint_xray_policy.json,
+      ],
+    var.vpc_arn == "" ? [] : [data.aws_iam_policy_document.networking_policy_document[0].json]),
+  }, var.policy_documents_to_attach)
+}
+
+data "aws_iam_policy_document" "iam-policy-document" {
+  for_each                = local.policy_documents_to_attach
+  source_policy_documents = each.value
```

Comment:
Do we need to worry about this?

> Statements defined in source_policy_documents must have unique `sid`s

Reply:
@ethanmills thanks for taking a look - I was going to reach out for input.

Yeah, that's a super valid point I hadn't considered. If we go ahead with this approach, I'll add a validation step in to ensure uniqueness.